src/caffe/layers/data_layer.cpp

```diff
@@ -35,11 +35,12 @@ OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
```

Comment:
Would you please articulate why you made so many changes to this class? I don't think all of them are needed for this particular feature, right?

Reply:
Yes, but these modifications do not conflict with the original Caffe. They can provide more powerful functions to meet our needs.

Reply:
How about submit another PR for the enhancements you think useful? Let's just focus on the regression support in this PR.

Reply:
Thank you for your concern, but I think I may have misunderstood my meaning. When using the multi label regression problem of image data, I met the direct LMDB data that did not support it. By modifying this class, that is, the LMDB data reading method, we solved this problem, not by converting it to h5py. From this improvement, we find that the effect is better than the traditional method and has a good performance on other similar problems, so we think this is an improvement on Intel Caffe and we hope to get your approval.

Reply:
Thank you for your reminding. We have made changes according to your opinion and submitted PR again. Here, I would like to explain to you our improvement. When we use the multi label regression problem of image data, I encounter the problem that LMDB database does not directly support. By modifying this class, that is, the LMDB data reading method, we solve this problem instead of converting it to H5Py. From this improvement, we find that the improvement is better than the traditional method and has a good performance on other similar problems, so we think this is an improvement on Intel coffee and we hope to get your approval.